### PR TITLE
Bump baggageclaim to v1.8.0 to fix deeply-nested volumes with overlay driver

### LINKED ISSUE #5799: Volumes can only be reused twice
## Summary

If you input and output the same volume in 3 or more tasks in the same job, the third task will fail with `failed to create volume`.

## Steps to reproduce

Set the following pipeline:

```yaml
resources:
- name: some-resource
  type: mock
  source:
    create_files:
      some_file: |
        hello you

jobs:
- name: reuse-volumes
  plan:
  - get: some-resource
  - task: step1
    config: &modify_some_file
      platform: linux

      inputs:
      - name: some-resource

      outputs:
      - name: some-resource

      image_resource:
        type: mock
        source: {mirror_self: true}

      run:
        path: sh
        args:
        - '-ec'
        - |
          echo "and you" >> some-resource/some_file
  - task: step2
    config: *modify_some_file
  - task: step3
    config: *modify_some_file
  - task: show-results
    config:
      platform: linux

      inputs:
      - name: some-resource

      image_resource:
        type: mock
        source: {mirror_self: true}

      run:
        path: sh
        args:
        - '-ec'
        - |
          cat some-resource/some_file
```

Trigger the `reuse-volumes` job.  The job will fail at task `step3`.  (I can PR this as a `testflight` case if there is interest.)

## Expected results

The job should complete successfully, and the `show-results` task should output

```
hello you
and you
and you
and you
```

## Actual results

Task `step3` fails with `failed to create volume`.

## Additional context

We observed this in a bigger pipeline.  Looking at the systemd journal on the box, we found the line:

    Jun 19 15:40:59 ip-10-1-22-245 kernel: [  625.625135] overlayfs: maximum fs stacking depth exceeded

It is not possible to stack overlayfs mounts indefinitely; there is a [hardcoded limit of 2](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/fs.h#n251) in the linux kernel.

I suspect (but have been unable as yet to positively confirm) that concourse/baggageclaim#35 is the cause of this issue.

## Triaging info

* Concourse version: 6.3.0
* Browser (if applicable): N/A
* Did this used to work?  Yes, we upgraded from 6.0.0 where we did not observe this problem.

### PR DESCRIPTION
**Pull request by vito**
[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## What does this PR accomplish?

**Bug Fix** | Feature | Documentation

closes #5799 

## Changes proposed by this PR:

Bumps Baggageclaim to [v1.8.0](https://github.com/concourse/baggageclaim/releases/tag/v1.8.0).

## Contributor Checklist

- [x] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [x] [Signed] all commits
- [ ] ~~Added tests (Unit and/or Integration)~~
- [ ] ~~Updated [Documentation]~~
- [ ] ~~Added release note (Optional)~~

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs

## Reviewer Checklist

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

**Comment by aoldershaw** [comment]
Looks like there's a merge conflict

File: go.sum [hunk 1]
```diff
@@ -18,6 +18,7 @@ code.cloudfoundry.org/credhub-cli v0.0.0-20190415201820-e3951663d25c h1:qdDzK71H
 code.cloudfoundry.org/credhub-cli v0.0.0-20190415201820-e3951663d25c/go.mod h1:oZAAU/4R7ssJ+kighWL4VLyyyY+Utm8H9r721AE3edQ=
 code.cloudfoundry.org/garden v0.0.0-20181108172608-62470dc86365 h1:W9nK4TUFQCzLniQN2xykFvzAteV0FogE94bVj8Ulu8A=
 code.cloudfoundry.org/garden v0.0.0-20181108172608-62470dc86365/go.mod h1:9edl55ou1627lC7chvtltZ561+iITkNTzkZsLgohU8s=
+code.cloudfoundry.org/lager v1.1.0 h1:v0RELJ2jqTeF2DW7PNjZaaGlrXbVxJBVz3uLxdP3fuY=
```
Comment (xtreme-sameer-vohra) [inline go.sum:21]: hmm, this is interesting

Comment (vito) [reply go.sum]: fyi: just rebased and now it only changes baggageclaim lines

**Comment by xtreme-sameer-vohra** [comment]
Tested the original failure case documented in #5799 
Thanks 👍

**Pull request merged**